site/app/controllers/OfficeHoursQueueController.php

```diff
@@ -67,146 +93,184 @@ public function addPerson() {
         }
+
+
+        $this->core->getQueries()->finishHelpUser($_POST['user_id'], $_POST['queue_code'], $remove_type);
         return Response::RedirectOnlyResponse(
             new RedirectResponse($this->core->buildCourseUrl(['office_hours_queue']))
         );
     }
 
     /**
-    * @param
     * @Route("/{_semester}/{_course}/office_hours_queue/empty", methods={"POST"})
     * @AccessControl(role="LIMITED_ACCESS_GRADER")
     * @return Response
     */
     public function emptyQueue() {
-        $this->core->getQueries()->emptyQueue($this->core->getUser()->getId());
+        if (!isset($_POST['queue_code']) && $_POST['queue_code'] != "") {
+            $this->core->addErrorMessage("Missing queue code");
+            return Response::RedirectOnlyResponse(
+                new RedirectResponse($this->core->buildCourseUrl(['office_hours_queue']))
+            );
+        }
+
+        $this->core->getQueries()->emptyQueue($_POST['queue_code']);
         return Response::RedirectOnlyResponse(
             new RedirectResponse($this->core->buildCourseUrl(['office_hours_queue']))
         );
     }
 
     /**
-    * @param
-    * @Route("/{_semester}/{_course}/office_hours_queue/remove", methods={"POST"})
+    * @Route("/{_semester}/{_course}/office_hours_queue/toggle", methods={"POST"})
+    * @AccessControl(role="LIMITED_ACCESS_GRADER")
     * @return Response
     */
-    public function removePerson() {
-        if (!isset($_POST['entry_id'])) {
-            $this->core->addErrorMessage("Missing entry ID");
+    public function toggleQueue() {
+        if (!isset($_POST['queue_code']) && $_POST['queue_code'] != "") {
+            $this->core->addErrorMessage("Missing queue code");
             return Response::RedirectOnlyResponse(
                 new RedirectResponse($this->core->buildCourseUrl(['office_hours_queue']))
             );
         }
-        if (!$this->core->getUser()->accessGrading() && $this->core->getUser()->getId() != $this->core->getQueries()->getUserIdFromQueueSlot($_POST['entry_id'])) {
-            $this->core->addErrorMessage("Permission denied to remove that person");
+        if (!isset($_POST['queue_state']) && $_POST['queue_state'] != "") {
+            $this->core->addErrorMessage("Missing queue state");
             return Response::RedirectOnlyResponse(
                 new RedirectResponse($this->core->buildCourseUrl(['office_hours_queue']))
             );
         }
-        $this->core->addSuccessMessage("Removed from queue");
-        $this->core->getQueries()->removeUserFromQueue($_POST['entry_id'], $this->core->getUser()->getId());
+
+        $this->core->getQueries()->toggleQueue($_POST['queue_code'], $_POST['queue_state']);
+        if ($_POST['queue_state'] == 0) {
+            $this->core->addSuccessMessage('Opened queue: "' . $_POST['queue_code'] . '"');
+        }
+        else {
+            $this->core->addSuccessMessage('Closed queue: "' . $_POST['queue_code'] . '"');
+        }
```

Comment:
Let's make this a ternary statement with inline templating:
`$this->core->addSuccessMessage(($_POST['queue_state'] === 0 ? 'Opened' : 'Closed') . ' queue: "{$_POST['queue_code']}"');`